packages/lit-dev-content/site/tutorials/content/tooltip/07.md

```diff
@@ -0,0 +1,58 @@
+You've polished the tooltip's behavior with animations. Now it's time to ensure
+great performance. Since a tooltip displays only when the user interacts with its
+target, it doesn't need to be created until then. If a page uses a lot of
+tooltips, which is not uncommon, adding this optimization can speed up initial
+page rendering by a meaningful amount.
+
+To do this, you'll create a function users will call with a tooltip target
+element and a callback that will render tooltip content on demand when
+its needed. Add a `static` method to the `SimpleTooltip` class called
+`lazy`. It will setup event listeners on the target element that listen on the
+same events used to trigger the tooltip. The listener creates the tooltip
+element and calls the user callback with it to fill in the content before
+showing it. Near the top of the class definition, add:
+
+```ts
+static lazy(target: Element, callback: (target: SimpleTooltip) => void) {
+  let called = false;
+  enterEvents.forEach(name => target.addEventListener(name, () => {
+    if (!called) {
+      called = true;
+      const tooltip = document.createElement('simple-tooltip') as SimpleTooltip;
+      callback(tooltip);
+      target.parentNode!.insertBefore(tooltip, target.nextSibling);
+      tooltip.show();
+    }
+  }, {once: true}));
+}
+```
+<aside class="info">The <code>once</code> option is used to automatically
+remove the event listener after it fires once. Since the callback should be
+called only once and there are multiple triggering events, a flag is used to
+manage the state.</aside>
+
+Now open the `my-content.ts` module and change one of the tooltips to use
+the `lazy` method. To do so, first add an `id` to the target and remove the
+inline tooltip:
+
+```html
+<p>
+  <span id="greeting">Hello, ${this.name}!</span>
+</p>
+```
+
+Then add a `firstUpdated` method to setup the call to `lazy`:
+
+```ts
+firstUpdated() {
```

Comment:
Forgot to mention the import of the SimpleTooltip class I suggest maybe this code snippet:

```ts
import {SimpleTooltip} from './simple-tooltip.js';

@customElement('my-content')
export class MyContent extends LitElement {
  ...
  firstUpdated() {
    const greeting = this.shadowRoot!.getElementById('greeting')!;
    SimpleTooltip.lazy(greeting, (tooltip: SimpleTooltip) => {
      tooltip.textContent = `${this.name}, there's coffee available in the lounge.`;
    });
  }
}
```

Reply:
Done